testsuite/feelcrb/test_deim.cpp

```diff
@@ -0,0 +1,151 @@
+#define BOOST_TEST_MODULE deim testsuite
+
+#include <testsuite/testsuite.hpp>
+
+#include <feel/feelalg/backend.hpp>
+#include <feel/feeldiscr/pch.hpp>
+#include <feel/feelfilters/loadmesh.hpp>
+#include <feel/feelvf/form.hpp>
+#include <feel/feelcrb/deim.hpp>
+#include <feel/feelcrb/parameterspace.hpp>
+
+using namespace Feel;
+
+inline
+po::options_description makeOptions()
+{
+    po::options_description options( "Test DEIM Options" );
+
+    options.add( feel_options() )
+        .add(eimOptions());
+    return options;
+}
+
+inline
+AboutData
+makeAbout()
+{
+    AboutData about( "test_deim" ,
+                     "test_deim" ,
+                     "0.1",
+                     "DEIM test",
+                     Feel::AboutData::License_GPL,
+                     "Copyright (c) 2017 Feel++ Consortium" );
+
+    return about;
+}
+
+class DeimTest :
+    public boost::enable_shared_from_this<DeimTest>
+{
+public :
+    typedef DeimTest self_type;
+
+    typedef double value_type;
+    typedef Mesh<Simplex<2> > mesh_type;
+    typedef boost::shared_ptr<mesh_type> mesh_ptrtype;
+
+    typedef Backend<double> backend_type;
+    typedef boost::shared_ptr<backend_type> backend_ptrtype;
+    typedef backend_type::vector_type vector_type;
+    typedef backend_type::vector_ptrtype vector_ptrtype;
+
+    typedef ParameterSpace<2> parameterspace_type;
+    typedef boost::shared_ptr<parameterspace_type> parameterspace_ptrtype;
+    typedef parameterspace_type::element_type parameter_type;
+
+    typedef DEIM<self_type> deim_type;
+    typedef boost::shared_ptr<deim_type> deim_ptrtype;
+
+    DeimTest() :
+        Dmu( parameterspace_type::New(2) )
+    {
+        auto mesh = loadMesh( _mesh=new mesh_type, _filename="test_deim.geo");
+        auto Xh = Pch<1>( mesh );
+        auto u = Xh->element();
+        V = backend()->newVector( Xh );
+        V1 = backend()->newVector( Xh );
+        V2 = backend()->newVector( Xh );
+        V3 = backend()->newVector( Xh );
+
+        auto f1 = form1( _test=Xh, _vector=V1 );
```

Comment:
Ah ok you use them for assembly later but this is quite ugly to have to do this like that. I will think of a better interface.